src/lock/runner_service.go

```diff
@@ -0,0 +1,76 @@
+package lock
+
+import (
+	"sync"
+
+	"github.com/divideandconquer/go-consul-client/src/config"
+	"github.com/promoboxx/go-discovery/src/discovery"
+)
+
+// RunnerServer is the interface to allow operations on a RunnerServer
+type RunnerServer interface {
+	Run() error
+	Stop() *sync.WaitGroup
+}
+
+type runner struct {
+	runner *Runner
+	state  string
+}
+
+func (r *runner) run() error {
+	err := r.runner.Run()
+	if err != nil {
+		return err
+	}
+	r.state = "running"
+	return nil
+}
+
+func (r *runner) stop() *sync.WaitGroup {
+	if r.state == "running" {
+		return r.runner.Stop()
+	}
+	return nil
+}
+
+type runnerServer struct {
+	environment string
+	serviceName string
+	conf        config.Loader
+	finder      discovery.Finder
+	tracer      Tracer
+	runners     []*runner
+	// client manager field here?
```

Comment:
@tsharkey since services like facebookpage require some clients, did you mean to have a client manager field here so we wouldn't need to implement any getClient functions?

Reply:
Yeah we will have to think about how we can work in these clients. I have a few ideas but I will have to think them through first